backend/notify-nl/src/main/kotlin/com/ritense/valtimoplugins/notifynl/service/NotifyNlTokenGenerationService.kt

```diff
@@ -26,17 +26,27 @@ import java.util.UUID
 private val logger = KotlinLogging.logger {}
 
 class NotifyNlTokenGenerationService {
-    fun generateToken(serviceId: UUID, secretKey: UUID): String {
+    fun generateFullToken(apiKey: String): String {
         logger.debug { "Generating a token for a request to NotifyNL" }
 
-        val signingKey = Keys.hmacShaKeyFor(secretKey.toString().toByteArray(Charset.forName("UTF-8")))
+        val regex = Regex(
```

Comment:
The regex approach  can use a comment to describe the purpose
something like:
// The api key consists of two part,......